storage/innobase/btr/btr0cur.cc

```diff
@@ -4605,10 +4605,6 @@ btr_cur_pessimistic_update(
 
 	rec = btr_cur_get_rec(cursor);
 
-	*offsets = rec_get_offsets(
-		rec, index, *offsets, page_is_leaf(page),
-		ULINT_UNDEFINED, offsets_heap);
-
```

Comment:
Good catch! This applies to 10.2 already. But, please let us assert that it is valid:
```c++
ut_ad(rec_offs_validate(rec, index, *offsets));
```